core/src/main/java/org/infinispan/stream/impl/LockStreamImpl.java

```diff
@@ -0,0 +1,231 @@
+      realStream.forEach(new CacheEntryConsumer<>(biConsumer, predicate));
+   }
+
+   @Override
+   public BaseCacheStream sequentialDistribution() {
+      return newOrReuse(realStream.sequentialDistribution());
+   }
+
+   @Override
+   public BaseCacheStream parallelDistribution() {
+      return newOrReuse(realStream.parallelDistribution());
+   }
+
+   @Override
+   public BaseCacheStream filterKeySegments(Set<Integer> segments) {
+      return newOrReuse(realStream.filterKeySegments(segments));
+   }
+
+   @Override
+   public BaseCacheStream filterKeys(Set<?> keys) {
+      return newOrReuse(realStream.filterKeys(keys));
+   }
+
+   @Override
+   public BaseCacheStream distributedBatchSize(int batchSize) {
+      return newOrReuse(realStream.distributedBatchSize(batchSize));
+   }
+
+   @Override
+   public BaseCacheStream segmentCompletionListener(SegmentCompletionListener listener) {
+      throw new UnsupportedOperationException("LockStream doesn't support completion listener");
+   }
+
+   @Override
+   public BaseCacheStream disableRehashAware() {
+      return newOrReuse(realStream.disableRehashAware());
+   }
+
+   @Override
+   public BaseCacheStream timeout(long timeout, TimeUnit unit) {
+      return newOrReuse(realStream.timeout(timeout, unit));
+   }
+
+   @Override
+   public Iterator<CacheEntry<K, V>> iterator() {
+      return realStream.iterator();
+   }
+
+   @Override
+   public Spliterator<CacheEntry<K, V>> spliterator() {
+      return realStream.spliterator();
+   }
+
+   @Override
+   public boolean isParallel() {
+      return realStream.isParallel();
+   }
+
+   @Override
+   public LockedStream<K, V> sequential() {
+      return newOrReuse(realStream.sequential());
+   }
+
+   @Override
+   public LockedStream<K, V> parallel() {
+      return newOrReuse(realStream.parallel());
+   }
+
+   @Override
+   public LockedStream<K, V> unordered() {
+      // This stream is always unordered
+      return this;
+   }
+
+   @Override
+   public LockedStream<K, V> onClose(Runnable closeHandler) {
+      return newOrReuse(realStream.onClose(closeHandler));
+   }
+
+   @Override
+   public void close() {
+      realStream.close();
+   }
+
+   @SerializeWith(value = CacheEntryConsumer.CacheEntryConsumerExternalizer.class)
+   private static class CacheEntryConsumer<K, V> implements BiConsumer<Cache<K, V>, CacheEntry<K, V>>, Serializable {
+      private final BiConsumer<Cache<K, V>, ? super CacheEntry<K, V>> realConsumer;
+      private final Predicate<? super CacheEntry<K, V>> predicate;
+      private transient LockManager lockManager;
+
+      private CacheEntryConsumer(BiConsumer<Cache<K, V>, ? super CacheEntry<K, V>> realConsumer,
+            Predicate<? super CacheEntry<K, V>> predicate) {
+         this.realConsumer = realConsumer;
+         this.predicate = predicate;
+      }
+
+      @Override
+      public void accept(Cache<K, V> kvCache, CacheEntry<K, V> kvCacheEntry) {
```

Comment:
Sorry I am missing the reference. If you mean the variables names, you can blame IntelliJ :D